scripts/apps/authoring-react/toolbar/highlights-management.tsx

```diff
@@ -0,0 +1,100 @@
+import {sdApi} from 'api';
+import {dispatchInternalEvent} from 'core/internal-events';
+import {Card} from 'core/ui/components/Card';
+import {Spacer} from 'core/ui/components/Spacer';
+import {gettext} from 'core/utils';
+import React from 'react';
+import {IArticle, IHighlight} from 'superdesk-api';
+import {Button, Heading, IconButton, Text} from 'superdesk-ui-framework/react';
+
+interface IProps {
+    article: IArticle;
+    closePopup(): void;
+}
+
+interface IStateLoading {
+    initialized: false;
+}
+
+interface IStateLoaded {
+    initialized: true;
+    selectedHighlighIds: Array<IHighlight> | null;
+}
+
+type IState = IStateLoaded | IStateLoading;
+
+export class HighlightsCardContent extends React.PureComponent<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            initialized: false,
+        };
+
+        this.unmarkHighlight = this.unmarkHighlight.bind(this);
+    }
+
+    componentDidMount(): void {
+        sdApi.highlights.fetchHighlights().then((res) => {
+            this.setState({
+                ...this.state,
+                initialized: true,
+                selectedHighlighIds: res._items
+                    .filter((highlight) => this.props.article.highlights.includes(highlight._id)),
+            });
+        });
+    }
+
+    unmarkHighlight(highlightId: string): void {
```

Comment:
It'd help to order component methods so it's easier to navigate. I order them the following way from bottom to top: `render` method, other react component life-cycle methods, custom methods and lastly the constructor at the top.